### ISSUE TITLE
Chrome raise `Failed to parse SourceMap` when I add `bless()` to gulp task

### ISSUE THREAD

**Issue by leomao10** [BlessCSS/gulp-bless#33]
Hi There,

When I tried to apply `bless` in our gulp task, we found that Chrome raise following error:

```
Failed to parse SourceMap: http://localhost:7001/app/maps/index.css.map
```

I tried to compare the difference between the source file it generated before and after gulp task and found that in origin file, source file format looks like this:

``` json
{
  "version":3,
  "file":"index.css",
  "sources":["index.css"],
  "mappings": "mapping string",
  "sourceRoot":"/source/",
  "sourcesContent":["source_content"]
}
```

Chrome do not complain for this format.

And source file generate after I added `bless()` looks like this:

``` json
{
  "version":3,
  "file":"index.css",
  "sources":["index.css"],
  "mappings": {
    "version":3,
    "sources":["full_path/index.css"],
    "names":[],
    "mappings":"mapping string",
    "sourcesContent":["source_content"]
  },
  "sourceRoot":"/source/",
  "sourcesContent":[]
}
```

Now, Chrome complain that it can't parse SourceMap even thought it is a valid json file. 

Not sure if it is because we did not use `gulp-bless` correctly. Here is the gulp task we use, and the version of `gulp bless` we use is "3.1.0":

``` javascript
'use strict';

var path = require('path');
var gulp = require('gulp');
var conf = require('./conf');
var bless = require('gulp-bless');

var browserSync = require('browser-sync');

var $ = require('gulp-load-plugins')();

var wiredep = require('wiredep').stream;
var _ = require('lodash');

gulp.task('styles', function () {
  var lessOptions = {
    options: [
      'bower_components',
      path.join(conf.paths.src, '/app')
    ]
  };

  var injectFiles = gulp.src([
    path.join(conf.paths.src, '/app/**/*.less'),
    path.join(conf.paths.src, '/fonts/*.less'),
    path.join('!' + conf.paths.src, '/app/index.less')
  ], { read: false });

  var injectOptions = {
    transform: function(filePath) {
      filePath = filePath.replace(conf.paths.src + '/app/', '');
      return '@import "' + filePath + '";';
    },
    starttag: '// injector',
    endtag: '// endinjector',
    addRootSlash: false
  };


  return gulp.src([
    path.join(conf.paths.src, '/app/index.less')
  ])
    .pipe($.inject(injectFiles, injectOptions))
    .pipe(wiredep(_.extend({}, conf.wiredep)))
    .pipe($.sourcemaps.init())
    .pipe($.less(lessOptions)).on('error', conf.errorHandler('Less'))
    .pipe($.autoprefixer()).on('error', conf.errorHandler('Autoprefixer'))
    .pipe(bless())
    .pipe($.sourcemaps.write('maps'))
    .pipe(gulp.dest(path.join(conf.paths.tmp, '/serve/app/')))
    .pipe(browserSync.reload({ stream: true }));
```

It would be really great if you can help to have a look and see how we can fix this issue.

Thank you in advance

**Comment by adam-lynch** [comment]
Hmm I'm not sure why that's happening. Could you try running a gulp task like this?

```javascript
gulp.task('debug', function(){
  return gulp.src([
      path.join(conf.paths.src, '/app/index.less')
    ])
      .pipe($.sourcemaps.init({ debug: true }))
      .pipe($.less(lessOptions)).on('error', conf.errorHandler('Less'))
      .pipe(bless())
      .pipe($.sourcemaps.write('maps', { debug: true }))
      .pipe(gulp.dest(path.join(conf.paths.tmp, '/serve/app/')));
});
```

Does it still not work? Is there any useful information logged to the console?

Also, try with `.pipe($.sourcemaps.init({ debug: true, identityMap: true }))`

**Comment by leomao10** [comment]
Hey @adam-lynch 

I need to keey these two lines:
```
    .pipe($.inject(injectFiles, injectOptions))
    .pipe(wiredep(_.extend({}, conf.wiredep)))
```
Otherwise, the task will failed when it process `$.less`.

Removing the line for `$.autoprefixer()` doesn't help, also `.pipe($.sourcemaps.init({ debug: true, identityMap: true }))` gave me the same output:

 ```
 gulp-sourcemap-write: No source content for "index.css". Loading from file.
gulp-sourcemap-write: source file not found: /Users/leo/reInteractive/pwc/pwc-enterprise/frontend/app/index.css
gulp-sourcemap-write: No source content for "index.css". Loading from file.
gulp-sourcemap-write: source file not found: /Users/leo/reInteractive/pwc/pwc-enterprise/frontend/app/index.css
```

What it complained is it can't find `index.css`, but I only has `index.less` file not `index.css` file.

**Comment by alvinlin123** [comment]
I may be talking nonsense here but from my brief research the mapping file produced by gulp-bless seems to have the wrong `mappings` values, it looks the plugin embedded a map inside the `mappings` field. I suspect the function [addSourceMap](https://github.com/BlessCSS/gulp-bless/blob/master/index.js#L56-L65) is passing in the wrong second argument?

I think we probably need to call the `applySourceMap` like
```
var map = result.maps[blessOutputIndex]; //this is already a valid mapping JSON 
 map['file'] = fileToAddTo.relative;
 applySourcemap(fileToAddTo, map);
```

I'm still fairly new to source mapping, but above is my suspicion based on some research :)

**Comment by adam-lynch** [comment]
@alvinlin123 wow, looks like you're right. See https://github.com/BlessCSS/gulp-bless/pull/34... I think that fixes it.

@leomao10 to test my changes on that branch you could `npm install --save-dev git://github.com/BlessCSS/gulp-bless.git#fixing_sourcemaps` and then run your gulp task as normal.

**Comment by alvinlin123** [comment]
The fix for this issue has been released to npm in version 3.2.1. Closing this issue.
